File: PatientAdministrationSystem.Api/PatientAdministrationSystem/Models/PatientWithVisits.cs
namespace Hci.Ah.Home.Api.Gateway.Models;

public class PatientWithVisits(Guid id, string firstName, string lastName, string email, IEnumerable<Visit> visits) : Patient(id, firstName, lastName, email)
{
    public IEnumerable<Visit> Visits { get; } = visits;
}

File: PatientAdministrationSystem.App/src/models/patientWithVisits.ts
import Patient from "./patient";
import Visit from "./visit";

export default class PatientWithVisits extends Patient {
    readonly visits: Visit[];

    constructor(id: string, firstName: string, lastName: string, email: string, visits: Visit[]) {
        super(id, firstName, lastName, email);
        this.visits = visits;
    }
}